packages/dd-trace/src/appsec/telemetry/waf.js

```diff
@@ -98,6 +109,39 @@ function incrementWafRequests (store) {
   }
 }
 
+function incrementTruncatedMetrics (metrics, truncationReason) {
+  const truncationTags = { truncation_reason: truncationReason }
+  appsecMetrics.count('waf.input_truncated', truncationTags).inc(1)
+
+  if (metrics?.maxTruncatedString) {
+    appsecMetrics.distribution('waf.truncated_value_size',
+      { truncation_reason: TRUNCATION_FLAGS.LONG_STRING })
+      .track(metrics.maxTruncatedString)
+  }
+
+  if (metrics?.maxTruncatedContainerSize) {
+    appsecMetrics.distribution('waf.truncated_value_size',
+      { truncation_reason: TRUNCATION_FLAGS.LARGE_CONTAINER })
+      .track(metrics.maxTruncatedContainerSize)
+  }
+
+  if (metrics?.maxTruncatedContainerDepth) {
+    appsecMetrics.distribution('waf.truncated_value_size',
+      { truncation_reason: TRUNCATION_FLAGS.DEEP_CONTAINER })
+      .track(metrics.maxTruncatedContainerDepth)
+  }
+}
+
+function getTruncationReason ({ maxTruncatedString, maxTruncatedContainerSize, maxTruncatedContainerDepth }) {
+  let reason = 0
+
+  if (maxTruncatedString) reason |= TRUNCATION_FLAGS.LONG_STRING
+  if (maxTruncatedContainerSize) reason |= TRUNCATION_FLAGS.LARGE_CONTAINER
+  if (maxTruncatedContainerDepth) reason |= TRUNCATION_FLAGS.DEEP_CONTAINER
```

Comment:
a small nit but unless you found those wordings somewhere in the RFC, i'd rather have consistent wording everywhere
```suggestion
  if (maxTruncatedString) reason |= TRUNCATION_FLAGS.STRING
  if (maxTruncatedContainerSize) reason |= TRUNCATION_FLAGS.CONTAINER_SIZE
  if (maxTruncatedContainerDepth) reason |= TRUNCATION_FLAGS.CONTAINER_DEPTH
```

Reply:
It's not mentioned in the RFC just my suggestion.
thanks for proposed names